src/scripts/scaling/scalingAction.sh

```diff
@@ -16,10 +14,328 @@ scaling_size=1
+    -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    -X GET \
+    $kubernetes_master/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions/domains.weblogic.oracle)
+  if [ $? -ne 0 ]
+    then
+      trace "Failed to retrieve Custom Resource Definition for WebLogic domain"
+      trace "CRD: $CRD"
+      exit 1
+  fi
+
+# Find domain version
+  local domain_api_version
+  if [ -x "$(command -v jq)" ]; then
+    domain_api_version=$(echo "${CRD}" | jq -r '.spec.version')
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+print(json.load(sys.stdin)["spec"]["version"])
+INPUT
+domain_api_version=`echo ${CRD} | python cmds-$$.py`
+  fi
+  echo "$domain_api_version"
+}
+
+# Retrieve Custom Resource Domain
+function get_custom_resource_domain() {
+  local DOMAIN=$(curl \
+    -v \
+    --cacert /var/run/secrets/kubernetes.io/serviceaccount/ca.crt \
+    -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+    $kubernetes_master/apis/weblogic.oracle/$domain_api_version/namespaces/$wls_domain_namespace/domains/$domain_uid)
+  if [ $? -ne 0 ]; then
+    trace "Failed to retrieve WebLogic Domain Custom Resource Definition"
+    exit 1
+  fi
+  echo "$DOMAIN"
+}
+
+# Verify if cluster is defined in clusters of the Custom Resource Domain
+# args:
+# $1 Custom Resource Domain
+function is_defined_in_clusters() {
+  local DOMAIN="$1"
+  local in_cluster_startup="False"
+  if [ -x "$(command -v jq)" ]; then
+    local inClusterStartupCmd="(.items[].spec.clusters[] | select (.clusterName == \"${wls_cluster_name}\"))"
+    local clusterDefinedInCRD=$(echo "${DOMAIN}" | jq "${inClusterStartupCmd}")
+    if [ "${clusterDefinedInCRD}" != "" ]; then
+      in_cluster_startup="True"
+    fi
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+outer_loop_must_break = False
+for i in json.load(sys.stdin)["items"]:
+  j = i["spec"]["clusters"]
+  for index, cs in enumerate(j):
+    if j[index]["clusterName"] == "$wls_cluster_name":
+      outer_loop_must_break = True
+      print True
+      break
+if outer_loop_must_break == False:
+  print False
+INPUT
+in_cluster_startup=`echo ${DOMAIN} | python cmds-$$.py`
+  fi
+  echo "$in_cluster_startup"
+}
+
+# Gets the current replica count of the cluster
+function get_num_ms_in_cluster() {
+  local num_ms
+  if [ -x "$(command -v jq)" ]; then
+  local numManagedServersCmd="(.items[].spec.clusters[] | select (.clusterName == \"${wls_cluster_name}\") | .replicas)"
+  num_ms=$(echo "${DOMAIN}" | jq "${numManagedServersCmd}")
+  else
+cat > cmds-$$.py << INPUT
+import sys, json
+for i in json.load(sys.stdin)["items"]:
+  j = i["spec"]["clusters"]
+  for index, cs in enumerate(j):
+    if j[index]["clusterName"] == "$wls_cluster_name":
+      print j[index]["replicas"]
+INPUT
+  num_ms=`echo ${DOMAIN} | python cmds-$$.py`
+  fi
+
+  if [ "${num_ms}" == "null" ]; then
+    num_ms=0
+  fi
+
+  echo "$num_ms"
```

Comment:
This should return 'min_replica_count' if min_replica_ count is higher than the discovered value.

Reply:
This function is only called from get_replica_count() function so made the change in one place to return 'min_replica_count' if min_replica_ count is higher than the discovered value.